Test failures with GEOS 3.12

The Debian package build fails with GEOS 3.12:
```
I: pybuild base:240: cd /build/geojson-pydantic-0.6.3/.pybuild/cpython3_3.11_geojson-pydantic/build; python3.11 -m pytest tests
============================= test session starts ==============================
platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.0.0+repack
rootdir: /build/geojson-pydantic-0.6.3/.pybuild/cpython3_3.11_geojson-pydantic/build
collected 134 items

tests/test_features.py ....................                              [ 14%]
tests/test_geometries.py ............FFFF....................F.......... [ 50%]
.F........F.......................................................       [ 99%]
tests/test_package.py .                                                  [100%]

=================================== FAILURES ===================================
_______________ test_multi_point_valid_coordinates[coordinates1] _______________

coordinates = [(1.0, 2.0)]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Empty array
            [],
            # No Z
            [(1.0, 2.0)],
            [(1.0, 2.0), (1.0, 2.0)],
            # Has Z
            [(1.0, 2.0, 3.0), (1.0, 2.0, 3.0)],
            # Mixed
            [(1.0, 2.0), (1.0, 2.0, 3.0)],
        ],
    )
    def test_multi_point_valid_coordinates(coordinates):
        """
        Two or three number elements as coordinates should be okay, as well as an empty array.
        """
        p = MultiPoint(type="MultiPoint", coordinates=coordinates)
        assert p.type == "MultiPoint"
        assert p.coordinates == coordinates
        assert hasattr(p, "__geo_interface__")
>       assert_wkt_equivalence(p)

tests/test_geometries.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = MultiPoint(type='MultiPoint', coordinates=[(1.0, 2.0)], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'MULTIPOINT ((1 2))' == 'MULTIPOINT (1 2)'
E         - MULTIPOINT (1 2)
E         + MULTIPOINT ((1 2))
E         ?            +     +

tests/test_geometries.py:25: AssertionError
_______________ test_multi_point_valid_coordinates[coordinates2] _______________

coordinates = [(1.0, 2.0), (1.0, 2.0)]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Empty array
            [],
            # No Z
            [(1.0, 2.0)],
            [(1.0, 2.0), (1.0, 2.0)],
            # Has Z
            [(1.0, 2.0, 3.0), (1.0, 2.0, 3.0)],
            # Mixed
            [(1.0, 2.0), (1.0, 2.0, 3.0)],
        ],
    )
    def test_multi_point_valid_coordinates(coordinates):
        """
        Two or three number elements as coordinates should be okay, as well as an empty array.
        """
        p = MultiPoint(type="MultiPoint", coordinates=coordinates)
        assert p.type == "MultiPoint"
        assert p.coordinates == coordinates
        assert hasattr(p, "__geo_interface__")
>       assert_wkt_equivalence(p)

tests/test_geometries.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = MultiPoint(type='MultiPoint', coordinates=[(1.0, 2.0), (1.0, 2.0)], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'MULTIPOINT ((1 2), (1 2))' == 'MULTIPOINT (1 2, 1 2)'
E         - MULTIPOINT (1 2, 1 2)
E         + MULTIPOINT ((1 2), (1 2))

tests/test_geometries.py:25: AssertionError
_______________ test_multi_point_valid_coordinates[coordinates3] _______________

coordinates = [(1.0, 2.0, 3.0), (1.0, 2.0, 3.0)]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Empty array
            [],
            # No Z
            [(1.0, 2.0)],
            [(1.0, 2.0), (1.0, 2.0)],
            # Has Z
            [(1.0, 2.0, 3.0), (1.0, 2.0, 3.0)],
            # Mixed
            [(1.0, 2.0), (1.0, 2.0, 3.0)],
        ],
    )
    def test_multi_point_valid_coordinates(coordinates):
        """
        Two or three number elements as coordinates should be okay, as well as an empty array.
        """
        p = MultiPoint(type="MultiPoint", coordinates=coordinates)
        assert p.type == "MultiPoint"
        assert p.coordinates == coordinates
        assert hasattr(p, "__geo_interface__")
>       assert_wkt_equivalence(p)

tests/test_geometries.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = MultiPoint(type='MultiPoint', coordinates=[(1.0, 2.0, 3.0), (1.0, 2.0, 3.0)], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'MULTIPOINT Z... 3), (1 2 3))' == 'MULTIPOINT Z (1 2 3, 1 2 3)'
E         - MULTIPOINT Z (1 2 3, 1 2 3)
E         + MULTIPOINT Z ((1 2 3), (1 2 3))

tests/test_geometries.py:25: AssertionError
_______________ test_multi_point_valid_coordinates[coordinates4] _______________

coordinates = [(1.0, 2.0), (1.0, 2.0, 3.0)]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Empty array
            [],
            # No Z
            [(1.0, 2.0)],
            [(1.0, 2.0), (1.0, 2.0)],
            # Has Z
            [(1.0, 2.0, 3.0), (1.0, 2.0, 3.0)],
            # Mixed
            [(1.0, 2.0), (1.0, 2.0, 3.0)],
        ],
    )
    def test_multi_point_valid_coordinates(coordinates):
        """
        Two or three number elements as coordinates should be okay, as well as an empty array.
        """
        p = MultiPoint(type="MultiPoint", coordinates=coordinates)
        assert p.type == "MultiPoint"
        assert p.coordinates == coordinates
        assert hasattr(p, "__geo_interface__")
>       assert_wkt_equivalence(p)

tests/test_geometries.py:74: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = MultiPoint(type='MultiPoint', coordinates=[(1.0, 2.0), (1.0, 2.0, 3.0)], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'MULTIPOINT Z...aN), (1 2 3))' == 'MULTIPOINT Z (1 2 0, 1 2 3)'
E         - MULTIPOINT Z (1 2 0, 1 2 3)
E         ?                   ^
E         + MULTIPOINT Z ((1 2 NaN), (1 2 3))
E         ?              +     ^^^^  +      +

tests/test_geometries.py:25: AssertionError
____________ test_multi_line_string_valid_coordinates[coordinates6] ____________

coordinates = [[(1.0, 2.0), (3.0, 4.0)], [(0.0, 0.0, 0.0), (1.0, 1.0, 1.0)]]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Empty array
            [],
            # One line, two points, no Z
            [[(1.0, 2.0), (3.0, 4.0)]],
            # One line, two points, has Z
            [[(0.0, 0.0, 0.0), (1.0, 1.0, 1.0)]],
            # One line, three points, no Z
            [[(1.0, 2.0), (3.0, 4.0), (5.0, 6.0)]],
            # Two lines, two points each, no Z
            [[(1.0, 2.0), (3.0, 4.0)], [(0.0, 0.0), (1.0, 1.0)]],
            # Two lines, two points each, has Z
            [[(1.0, 2.0, 0.0), (3.0, 4.0, 1.0)], [(0.0, 0.0, 0.0), (1.0, 1.0, 1.0)]],
            # Mixed
            [[(1.0, 2.0), (3.0, 4.0)], [(0.0, 0.0, 0.0), (1.0, 1.0, 1.0)]],
        ],
    )
    def test_multi_line_string_valid_coordinates(coordinates):
        """
        A list of two coordinates or more should be okay
        """
        multilinestring = MultiLineString(type="MultiLineString", coordinates=coordinates)
        assert multilinestring.type == "MultiLineString"
        assert multilinestring.coordinates == coordinates
        assert hasattr(multilinestring, "__geo_interface__")
>       assert_wkt_equivalence(multilinestring)

tests/test_geometries.py:148: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = MultiLineString(type='MultiLineString', coordinates=[[(1.0, 2.0), (3.0, 4.0)], [(0.0, 0.0, 0.0), (1.0, 1.0, 1.0)]], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'MULTILINESTR... 0 0, 1 1 1))' == 'MULTILINESTR... 0 0, 1 1 1))'
E         - MULTILINESTRING Z ((1 2 0, 3 4 0), (0 0 0, 1 1 1))
E         + MULTILINESTRING Z ((1 2 NaN, 3 4 NaN), (0 0 0, 1 1 1))

tests/test_geometries.py:25: AssertionError
____________________ test_polygon_with_holes[coordinates2] _____________________

coordinates = [[(0.0, 0.0), (0.0, 10.0), (10.0, 10.0), (10.0, 0.0), (0.0, 0.0)], [(2.0, 2.0, 2.0), (2.0, 4.0, 0.0), (4.0, 4.0, 0.0), (4.0, 2.0, 0.0), (2.0, 2.0, 2.0)]]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Polygon with holes, no Z
            [
                [(0.0, 0.0), (0.0, 10.0), (10.0, 10.0), (10.0, 0.0), (0.0, 0.0)],
                [(2.0, 2.0), (2.0, 4.0), (4.0, 4.0), (4.0, 2.0), (2.0, 2.0)],
            ],
            # Polygon with holes, has Z
            [
                [
                    (0.0, 0.0, 0.0),
                    (0.0, 10.0, 0.0),
                    (10.0, 10.0, 0.0),
                    (10.0, 0.0, 0.0),
                    (0.0, 0.0, 0.0),
                ],
                [
                    (2.0, 2.0, 1.0),
                    (2.0, 4.0, 1.0),
                    (4.0, 4.0, 1.0),
                    (4.0, 2.0, 1.0),
                    (2.0, 2.0, 1.0),
                ],
            ],
            # Mixed
            [
                [(0.0, 0.0), (0.0, 10.0), (10.0, 10.0), (10.0, 0.0), (0.0, 0.0)],
                [
                    (2.0, 2.0, 2.0),
                    (2.0, 4.0, 0.0),
                    (4.0, 4.0, 0.0),
                    (4.0, 2.0, 0.0),
                    (2.0, 2.0, 2.0),
                ],
            ],
        ],
    )
    def test_polygon_with_holes(coordinates):
        """Check interior and exterior rings."""
        polygon = Polygon(type="Polygon", coordinates=coordinates)
        assert polygon.type == "Polygon"
        assert hasattr(polygon, "__geo_interface__")
        assert polygon.exterior == polygon.coordinates[0]
        assert list(polygon.interiors) == [polygon.coordinates[1]]
>       assert_wkt_equivalence(polygon)

tests/test_geometries.py:234: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = Polygon(type='Polygon', coordinates=[[(0.0, 0.0), (0.0, 10.0), (10.0, 10.0), (10.0, 0.0), (0.0, 0.0)], [(2.0, 2.0, 2.0), (2.0, 4.0, 0.0), (4.0, 4.0, 0.0), (4.0, 2.0, 0.0), (2.0, 2.0, 2.0)]], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'POLYGON Z ((... 2 0, 2 2 2))' == 'POLYGON Z ((... 2 0, 2 2 2))'
E         - POLYGON Z ((0 0 0, 0 10 0, 10 10 0, 10 0 0, 0 0 0), (2 2 2, 2 4 0, 4 4 0, 4 2 0, 2 2 2))
E         + POLYGON Z ((0 0 NaN, 0 10 NaN, 10 10 NaN, 10 0 NaN, 0 0 NaN), (2 2 2, 2 4 0, 4 4 0, 4 2 0, 2 2 2))

tests/test_geometries.py:25: AssertionError
_______________________ test_multi_polygon[coordinates3] _______________________

coordinates = [[[(0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 1.0), (0.0, 0.0)], [(2.1, 2.1, 2.1), (2.2, 2.1, 2.0), (2.2, 2.2, 2.2), (2.1, 2.2, 2.3), (2.1, 2.1, 2.1)]]]

    @pytest.mark.parametrize(
        "coordinates",
        [
            # Empty array
            [],
            # Multipolygon, no Z
            [
                [
                    [(0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 1.0), (0.0, 0.0)],
                    [(2.1, 2.1), (2.2, 2.1), (2.2, 2.2), (2.1, 2.2), (2.1, 2.1)],
                ]
            ],
            # Multipolygon, has Z
            [
                [
                    [
                        (0.0, 0.0, 4.0),
                        (1.0, 0.0, 4.0),
                        (1.0, 1.0, 4.0),
                        (0.0, 1.0, 4.0),
                        (0.0, 0.0, 4.0),
                    ],
                    [
                        (2.1, 2.1, 4.0),
                        (2.2, 2.1, 4.0),
                        (2.2, 2.2, 4.0),
                        (2.1, 2.2, 4.0),
                        (2.1, 2.1, 4.0),
                    ],
                ]
            ],
            # Mixed
            [
                [
                    [(0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 1.0), (0.0, 0.0)],
                    [
                        (2.1, 2.1, 2.1),
                        (2.2, 2.1, 2.0),
                        (2.2, 2.2, 2.2),
                        (2.1, 2.2, 2.3),
                        (2.1, 2.1, 2.1),
                    ],
                ]
            ],
        ],
    )
    def test_multi_polygon(coordinates):
        """Should accept sequence of polygons."""
        multi_polygon = MultiPolygon(type="MultiPolygon", coordinates=coordinates)
    
        assert multi_polygon.type == "MultiPolygon"
        assert hasattr(multi_polygon, "__geo_interface__")
>       assert_wkt_equivalence(multi_polygon)

tests/test_geometries.py:311: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

geom = MultiPolygon(type='MultiPolygon', coordinates=[[[(0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 1.0), (0.0, 0.0)], [(2.1, 2.1, 2.1), (2.2, 2.1, 2.0), (2.2, 2.2, 2.2), (2.1, 2.2, 2.3), (2.1, 2.1, 2.1)]]], bbox=None)

    def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
        """Assert WKT equivalence with Shapely."""
        # Remove any trailing `.0` to match Shapely format
        clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
>       assert shape(geom).wkt == clean_wkt
E       AssertionError: assert 'MULTIPOLYGON....1 2.1 2.1)))' == 'MULTIPOLYGON....1 2.1 2.1)))'
E         - MULTIPOLYGON Z (((0 0 0, 1 0 0, 1 1 0, 0 1 0, 0 0 0), (2.1 2.1 2.1, 2.2 2.1 2, 2.2 2.2 2.2, 2.1 2.2 2.3, 2.1 2.1 2.1)))
E         ?                               -----  ^^^^^^^^^^^^^^
E         + MULTIPOLYGON Z (((0 0 NaN, 1 0 NaN, 1 1 NaN, 0 1 NaN, 0 0 NaN), (2.1 2.1 2.1, 2.2 2.1 2, 2.2 2.2 2.2, 2.1 2.2 2.3, 2.1 2.1 2.1)))
E         ?                       +++++++ ++++    +++++++  +++++++   ^^^^

tests/test_geometries.py:25: AssertionError
=========================== short test summary info ============================
FAILED tests/test_geometries.py::test_multi_point_valid_coordinates[coordinates1]
FAILED tests/test_geometries.py::test_multi_point_valid_coordinates[coordinates2]
FAILED tests/test_geometries.py::test_multi_point_valid_coordinates[coordinates3]
FAILED tests/test_geometries.py::test_multi_point_valid_coordinates[coordinates4]
FAILED tests/test_geometries.py::test_multi_line_string_valid_coordinates[coordinates6]
FAILED tests/test_geometries.py::test_polygon_with_holes[coordinates2] - Asse...
FAILED tests/test_geometries.py::test_multi_polygon[coordinates3] - Assertion...
======================== 7 failed, 127 passed in 0.79s =========================
E: pybuild pybuild:388: test: plugin pyproject failed with: exit code=1: cd /build/geojson-pydantic-0.6.3/.pybuild/cpython3_3.11_geojson-pydantic/build; python3.11 -m pytest tests
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ Note: Minor version `0.X.0` update might break the API, It's recommanded to pin 
 ### Fixed
 
 * reduce validation error message verbosity when discriminating Geometry types
+* MultiPoint WKT now includes parentheses around each Point
 
 ### Added
 

--- a/geojson_pydantic/geometries.py
+++ b/geojson_pydantic/geometries.py
@@ -67,7 +67,7 @@ def _lines_has_z(lines: List[LineStringCoords]) -> bool:
 def _polygons_wkt_coordinates(
     coordinates: List[PolygonCoords], force_z: bool = False
 ) -> str:
-    return ",".join(
+    return ", ".join(
         f"({_lines_wtk_coordinates(polygon, force_z)})" for polygon in coordinates
     )
 
@@ -134,7 +134,11 @@ class MultiPoint(_GeometryBase):
 
     def __wkt_coordinates__(self, coordinates: Any, force_z: bool) -> str:
         """return WKT coordinates."""
-        return _position_list_wkt_coordinates(coordinates, force_z)
+        # return _position_list_wkt_coordinates(coordinates, force_z)
+        return ", ".join(
+            f"({_position_wkt_coordinates(position, force_z)})"
+            for position in coordinates
+        )
 
     @property
     def has_z(self) -> bool:
@@ -319,7 +323,7 @@ Geometry = Annotated[
     Field(discriminator="type"),
 ]
 
-GeometryCollection.update_forward_refs()
+GeometryCollection.model_rebuild()
 
 
 def parse_geometry_obj(obj: Any) -> Geometry:

--- a/geojson_pydantic/geometries.py
+++ b/geojson_pydantic/geometries.py
@@ -134,7 +134,6 @@ class MultiPoint(_GeometryBase):
 
     def __wkt_coordinates__(self, coordinates: Any, force_z: bool) -> str:
         """return WKT coordinates."""
-        # return _position_list_wkt_coordinates(coordinates, force_z)
         return ", ".join(
             f"({_position_wkt_coordinates(position, force_z)})"
             for position in coordinates

--- a/tests/test_geometries.py
+++ b/tests/test_geometries.py
@@ -1,12 +1,8 @@
-import re
-from typing import Union
-
 import pytest
+import shapely
 from pydantic import ValidationError
-from shapely.geometry import shape
 
 from geojson_pydantic.geometries import (
-    Geometry,
     GeometryCollection,
     LineString,
     MultiLineString,
@@ -35,13 +31,6 @@ def test_pydantic_schema(obj):
     assert obj.model_json_schema()
 
 
-def assert_wkt_equivalence(geom: Union[Geometry, GeometryCollection]):
-    """Assert WKT equivalence with Shapely."""
-    # Remove any trailing `.0` to match Shapely format
-    clean_wkt = re.sub(r"\.0(\D)", r"\1", geom.wkt)
-    assert shape(geom).wkt == clean_wkt
-
-
 @pytest.mark.parametrize("coordinates", [(1.01, 2.01), (1.0, 2.0, 3.0), (1.0, 2.0)])
 def test_point_valid_coordinates(coordinates):
     """
@@ -51,7 +40,6 @@ def test_point_valid_coordinates(coordinates):
     assert p.type == "Point"
     assert p.coordinates == coordinates
     assert hasattr(p, "__geo_interface__")
-    assert_wkt_equivalence(p)
 
 
 @pytest.mark.parametrize(
@@ -88,7 +76,6 @@ def test_multi_point_valid_coordinates(coordinates):
     assert p.type == "MultiPoint"
     assert p.coordinates == coordinates
     assert hasattr(p, "__geo_interface__")
-    assert_wkt_equivalence(p)
 
 
 @pytest.mark.parametrize(
@@ -123,7 +110,6 @@ def test_line_string_valid_coordinates(coordinates):
     assert linestring.type == "LineString"
     assert linestring.coordinates == coordinates
     assert hasattr(linestring, "__geo_interface__")
-    assert_wkt_equivalence(linestring)
 
 
 @pytest.mark.parametrize("coordinates", [None, "Foo", [], [(1.0, 2.0)], ["Foo", "Bar"]])
@@ -162,7 +148,6 @@ def test_multi_line_string_valid_coordinates(coordinates):
     assert multilinestring.type == "MultiLineString"
     assert multilinestring.coordinates == coordinates
     assert hasattr(multilinestring, "__geo_interface__")
-    assert_wkt_equivalence(multilinestring)
 
 
 @pytest.mark.parametrize(
@@ -200,7 +185,6 @@ def test_polygon_valid_coordinates(coordinates):
     else:
         assert polygon.exterior is None
     assert not list(polygon.interiors)
-    assert_wkt_equivalence(polygon)
 
 
 @pytest.mark.parametrize(
@@ -248,7 +232,6 @@ def test_polygon_with_holes(coordinates):
     assert hasattr(polygon, "__geo_interface__")
     assert polygon.exterior == polygon.coordinates[0]
     assert list(polygon.interiors) == [polygon.coordinates[1]]
-    assert_wkt_equivalence(polygon)
 
 
 @pytest.mark.parametrize(
@@ -325,7 +308,6 @@ def test_multi_polygon(coordinates):
 
     assert multi_polygon.type == "MultiPolygon"
     assert hasattr(multi_polygon, "__geo_interface__")
-    assert_wkt_equivalence(multi_polygon)
 
 
 @pytest.mark.parametrize(
@@ -494,7 +476,6 @@ def test_geometry_collection_iteration(coordinates):
         type="GeometryCollection", geometries=[polygon, multipolygon]
     )
     assert hasattr(gc, "__geo_interface__")
-    assert_wkt_equivalence(gc)
     iter(gc)
 
 
@@ -508,7 +489,6 @@ def test_len_geometry_collection(coordinates):
     gc = GeometryCollection(
         type="GeometryCollection", geometries=[polygon, multipolygon]
     )
-    assert_wkt_equivalence(gc)
     assert len(gc) == 2
 
 
@@ -522,7 +502,6 @@ def test_getitem_geometry_collection(coordinates):
     gc = GeometryCollection(
         type="GeometryCollection", geometries=[polygon, multipolygon]
     )
-    assert_wkt_equivalence(gc)
     assert polygon == gc[0]
     assert multipolygon == gc[1]
 
@@ -530,13 +509,19 @@ def test_getitem_geometry_collection(coordinates):
 def test_wkt_mixed_geometry_collection():
     point = Point(type="Point", coordinates=(0.0, 0.0, 0.0))
     line_string = LineString(type="LineString", coordinates=[(0.0, 0.0), (1.0, 1.0)])
-    gc = GeometryCollection(type="GeometryCollection", geometries=[point, line_string])
-    assert_wkt_equivalence(gc)
+    assert (
+        GeometryCollection(
+            type="GeometryCollection", geometries=[point, line_string]
+        ).wkt
+        == "GEOMETRYCOLLECTION Z (POINT Z (0.0 0.0 0.0), LINESTRING (0.0 0.0, 1.0 1.0))"
+    )
 
 
 def test_wkt_empty_geometry_collection():
-    gc = GeometryCollection(type="GeometryCollection", geometries=[])
-    assert_wkt_equivalence(gc)
+    assert (
+        GeometryCollection(type="GeometryCollection", geometries=[]).wkt
+        == "GEOMETRYCOLLECTION EMPTY"
+    )
 
 
 def test_geometry_collection_warnings():
@@ -734,3 +719,45 @@ def test_wkt_empty_geometrycollection():
     assert GeometryCollection(type="GeometryCollection", geometries=[]).wkt.endswith(
         " EMPTY"
     )
+
+
+@pytest.mark.parametrize(
+    "wkt",
+    (
+        "POINT (0.0 0.0)",
+        # "POINT EMPTY" does not result in valid GeoJSON
+        "POINT Z (0.0 0.0 0.0)",
+        "MULTIPOINT ((0.0 0.0))",
+        "MULTIPOINT Z ((0.0 0.0 0.0))",
+        "MULTIPOINT ((0.0 0.0), (1.0 1.0))",
+        "MULTIPOINT Z ((0.0 0.0 0.0), (1.0 1.0 1.0))",
+        "MULTIPOINT EMPTY",
+        "LINESTRING (0.0 0.0, 1.0 1.0, 2.0 2.0)",
+        "LINESTRING Z (0.0 0.0 0.0, 1.0 1.0 1.0, 2.0 2.0 2.0)",
+        # "LINESTRING EMPTY" does not result in valid GeoJSON
+        "MULTILINESTRING ((0.0 0.0, 1.0 1.0))",
+        "MULTILINESTRING ((0.0 0.0, 1.0 1.0), (1.0 1.0, 2.0 2.0))",
+        "MULTILINESTRING Z ((0.0 0.0 0.0, 1.0 1.0 1.0))",
+        "MULTILINESTRING Z ((0.0 0.0 0.0, 1.0 1.0 1.0), (1.0 1.0 1.0, 2.0 2.0 2.0))",
+        "MULTILINESTRING EMPTY",
+        "POLYGON ((0.0 0.0, 1.0 1.0, 2.0 2.0, 3.0 3.0, 0.0 0.0))",
+        "POLYGON ((0.0 0.0, 4.0 0.0, 4.0 4.0, 0.0 4.0, 0.0 0.0), (1.0 1.0, 1.0 2.0, 2.0 2.0, 2.0 1.0, 1.0 1.0))",
+        "POLYGON Z ((0.0 0.0 0.0, 1.0 1.0 0.0, 2.0 2.0 0.0, 3.0 3.0 0.0, 0.0 0.0 0.0))",
+        "POLYGON Z ((0.0 0.0 0.0, 4.0 0.0 0.0, 4.0 4.0 0.0, 0.0 4.0 0.0, 0.0 0.0 0.0), (1.0 1.0 0.0, 1.0 2.0 0.0, 2.0 2.0 0.0, 2.0 1.0 0.0, 1.0 1.0 0.0))",
+        "POLYGON EMPTY",
+        "MULTIPOLYGON (((0.0 0.0, 1.0 1.0, 2.0 2.0, 3.0 3.0, 0.0 0.0)))",
+        "MULTIPOLYGON (((0.0 0.0, 1.0 1.0, 2.0 2.0, 3.0 3.0, 0.0 0.0)), ((1.0 1.0, 2.0 2.0, 3.0 3.0, 4.0 4.0, 1.0 1.0)))",
+        "MULTIPOLYGON Z (((0.0 0.0 0.0, 1.0 1.0 0.0, 2.0 2.0 0.0, 3.0 3.0 0.0, 0.0 0.0 0.0)))",
+        "MULTIPOLYGON Z (((0.0 0.0 0.0, 1.0 1.0 0.0, 2.0 2.0 0.0, 3.0 3.0 0.0, 0.0 0.0 0.0)), ((1.0 1.0 0.0, 2.0 2.0 0.0, 3.0 3.0 0.0, 4.0 4.0 0.0, 1.0 1.0 0.0)))",
+        "MULTIPOLYGON EMPTY",
+        "GEOMETRYCOLLECTION (POINT (0.0 0.0))",
+        "GEOMETRYCOLLECTION (POINT (0.0 0.0), MULTIPOINT ((0.0 0.0), (1.0 1.0)))",
+        "GEOMETRYCOLLECTION Z (POLYGON EMPTY, MULTIPOLYGON Z (((0.0 0.0 0.0, 1.0 1.0 0.0, 2.0 2.0 0.0, 3.0 3.0 0.0, 0.0 0.0 0.0))))",
+        "GEOMETRYCOLLECTION Z (LINESTRING Z (0.0 0.0 0.0, 1.0 1.0 1.0, 2.0 2.0 2.0), MULTILINESTRING ((0.0 0.0, 1.0 1.0), (1.0 1.0, 2.0 2.0)))",
+        "GEOMETRYCOLLECTION EMPTY",
+    ),
+)
+def test_wkt(wkt: str):
+    # Use Shapely to parse the input WKT so we know it is parsable by other tools.
+    # Then load it into a Geometry and ensure the output WKT is the same as the input.
+    assert parse_geometry_obj(shapely.from_wkt(wkt).__geo_interface__).wkt == wkt

